lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery/Random.hs

```diff
@@ -15,28 +17,57 @@ module Cardano.Wallet.Primitive.AddressDiscovery.Random
     (
     -- ** State
       RndState (..)
-    , mkRndState
     ) where
 
 import Prelude
 
+import Cardano.Wallet.Primitive.AddressDerivation
+    ( Depth (..), Key (..), XPrv )
+import Cardano.Wallet.Primitive.AddressDerivation.Random
+    ( decryptDerPath, eskToHdPassphrase )
 import Cardano.Wallet.Primitive.AddressDiscovery
     ( CompareDiscovery (..)
     , GenChange (..)
     , IsOurs (..)
     , IsOwned (..)
     , KnownAddresses (..)
     )
-import Data.ByteString
-    ( ByteString )
+import Control.DeepSeq
+    ( NFData )
+import Crypto.Error
+    ( CryptoFailable (..) )
+import Data.Map.Strict
+    ( Map )
+import Data.Text
+    ( Text )
+import GHC.Generics
+    ( Generic )
 
-newtype RndState = RndState ByteString
+import qualified Data.Map.Strict as Map
 
-mkRndState :: RndState
-mkRndState = error "AddressDiscovery.Random unimplemented"
+-- | Unified identifier for wallets (both FO and EO ones).
+-- Technically it contains a text, but actually it contains one from two options:
+--
+-- 1. for FO-wallets - 'Address' made from wallet's root public key, in Base58-format.
+-- 2. for EO-wallets - 'UUID', in Base58-format.
+newtype HdRootId = HdRootId { getHdRootId :: Text }
```

Comment:
Should we specialise the code to FO initially, then expand to EO when we implement that user story?

Reply:
it is just a comment explaining rationale behind Text

Reply:
We shouldn't anything about EO wallets at the moment. We haven't started anything regarding those (and won't before a while). Also, I don't understand the need for such type :thinking: ? (will probably find the answer further in the review..)

Reply:
removed as RndState changed and now contains only root private key